cmd/cloud/cluster.go

```diff
@@ -28,155 +27,29 @@ const (
-	clusterResizeCmd.Flags().Int("max-scaling", maxScalingDefault, "The maximum number of nodes to rotate every time. If the number is bigger than the number of nodes, then the number of nodes will be the maximum number.")
-	clusterResizeCmd.Flags().Int("max-drain-retries", maxDrainRetriesDefault, "The number of times to retry a node drain.")
-	clusterResizeCmd.Flags().Int("evict-grace-period", evictGracePeriodDefault, "The pod eviction grace period when draining in seconds.")
-	clusterResizeCmd.Flags().Int("wait-between-rotations", waitBetweenRotationsDefault, "Τhe time in seconds to wait between each rotation of a group of nodes.")
-	clusterResizeCmd.Flags().Int("wait-between-drains", waitBetweenDrainsDefault, "The time in seconds to wait between each node drain in a group of nodes.")
-	clusterResizeCmd.Flags().Int("wait-between-pod-evictions", waitBetweenPodEvictionsDefault, "The time in seconds to wait between each pod eviction in a node drain.")
-	clusterResizeCmd.MarkFlagRequired("cluster")
-
-	clusterDeleteCmd.Flags().String("cluster", "", "The id of the cluster to be deleted.")
-	clusterDeleteCmd.MarkFlagRequired("cluster")
-
-	clusterGetCmd.Flags().String("cluster", "", "The id of the cluster to be fetched.")
-	clusterGetCmd.MarkFlagRequired("cluster")
-
-	registerPagingFlags(clusterListCmd)
-	registerTableOutputFlags(clusterListCmd)
-
-	clusterUtilitiesCmd.Flags().String("cluster", "", "The id of the cluster whose utilities are to be fetched.")
-	clusterUtilitiesCmd.MarkFlagRequired("cluster")
-
-	clusterCmd.AddCommand(clusterCreateCmd)
-	clusterCmd.AddCommand(clusterProvisionCmd)
-	clusterCmd.AddCommand(clusterUpdateCmd)
-	clusterCmd.AddCommand(clusterUpgradeCmd)
-	clusterCmd.AddCommand(clusterResizeCmd)
-	clusterCmd.AddCommand(clusterDeleteCmd)
-	clusterCmd.AddCommand(clusterGetCmd)
-	clusterCmd.AddCommand(clusterListCmd)
-	clusterCmd.AddCommand(clusterInstallationCmd)
-	clusterCmd.AddCommand(clusterShowStateReport)
-	clusterCmd.AddCommand(clusterUtilitiesCmd)
-	clusterCmd.AddCommand(clusterShowSizeDictionary)
-	clusterCmd.AddCommand(clusterAnnotationCmd)
-}
+func newCmdCluster() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "cluster",
+		Short: "Manipulate clusters managed by the provisioning server.",
+	}
 
-var clusterCmd = &cobra.Command{
-	Use:   "cluster",
-	Short: "Manipulate clusters managed by the provisioning server.",
+	cmd.AddCommand(newCmdClusterCreate())
+	cmd.AddCommand(newCmdClusterProvision())
+	cmd.AddCommand(newCmdClusterUpdate())
+	cmd.AddCommand(newCmdClusterUpgrade())
+	cmd.AddCommand(newCmdClusterResize())
+	cmd.AddCommand(newCmdClusterDelete())
+	cmd.AddCommand(newCmdClusterGet())
+	cmd.AddCommand(newCmdClusterList())
+	cmd.AddCommand(newCmdClusterUtilities())
+
+	cmd.AddCommand(newCmdClusterSizeDictionary())
+	cmd.AddCommand(newCmdClusterShowStateReport())
+
+	cmd.AddCommand(clusterAnnotationCmd)
```

Comment:
What happen with this two commands?

Reply:
`dictionary` prints `predefined cluster size templates`. Like - 
```
"SizeAlef1000": {
    "MasterInstanceType": "t3.large",
    "MasterCount": 1,
    "NodeInstanceType": "m5.large",
    "NodeMinCount": 4,
    "NodeMaxCount": 4
},
```

And, `state-report` shows possible valid changes from each state.
```
"RequestedState": "creation-requested",
"ValidStates": [
    "creation-requested",
    "creation-failed"
],
```

Reply:
What I meant is, why did you didn't refactor `clusterAnnotationCmd` & `clusterAnnotationCmd`.

Reply:
`cloud cluster annotation` and `cloud cluster installation` both have sub-commands. So, in order to keep this PR as small as possible, I didn't touch these. I'll make a separate pull requesr.

Reply:
Gotcha! That sounds better, yes.